algorithms/kernel/dtrees/gbt/gbt_train_dense_default_impl.i

```diff
@@ -469,24 +481,79 @@ services::Status  computeTypeDisp(HostAppIface* pHostApp, const NumericTable *x,
             deleteTables<cpu>(aTbl, aTblImp, aTblSmplCnt, nTrees);
             break;
         }
+
         for(iTree = 0; iTree < nTrees; ++iTree)
+        {
+            if ((ptrTotalCover != nullptr) || (ptrCover != nullptr))
+            {
+                totalCoverFeature = (aTbl[iTree]->getArrayCoverFeature());
+            }
+            if ((ptrWeight != nullptr) || (ptrCover != nullptr) || (ptrGain != nullptr))
+            {
+                weightFeature = aTbl[iTree]->getArrayNumSplitFeature();
+            }
+            if ((ptrTotalGain != nullptr) || (ptrGain != nullptr))
+            {
+                totalGainFeature = (aTbl[iTree]->getArrayGainFeature());
+            }
+
+            if (ptrWeight != nullptr)
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    ptrWeight[kFeature] += static_cast<algorithmFPType>(weightFeature[kFeature]);
+
+            if (ptrTotalCover != nullptr)
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    ptrTotalCover[kFeature] += static_cast<algorithmFPType>(totalCoverFeature[kFeature]);
+
+            if (ptrTotalGain != nullptr)
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    ptrTotalGain[kFeature] += static_cast<algorithmFPType>(totalGainFeature[kFeature]);
+
+            if (ptrCover != nullptr)
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    ptrCover[kFeature] += static_cast<algorithmFPType>(totalCoverFeature[kFeature]);
+
+            if (ptrGain != nullptr)
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    ptrGain[kFeature] += static_cast<algorithmFPType>(totalGainFeature[kFeature]);
+
+            if ((ptrWeight != nullptr) || (ptrCover != nullptr) || (ptrGain != nullptr))
+                for(size_t kFeature = 0; kFeature < nStor; ++kFeature)
+                    allWeight[kFeature] += static_cast<algorithmFPType>(weightFeature[kFeature]);
+
             md.add(aTbl[iTree], aTblImp[iTree], aTblSmplCnt[iTree]);
+        }
+
         if((i + 1 < par.maxIterations) && task.done())
             break;
     }
 
+    for (size_t i = 0; i < nStor; ++i)
+    {
+        if ((ptrCover != nullptr) && (allWeight[i] != 0))
```

Comment:
why don`t we check 'ptrCover' on nullptr only once outside?

Reply:
Fixed